spectrum/checks.py

```diff
@@ -669,6 +669,35 @@ def recommend(self, manuscript='', subject='', keywords='', abstract=''):
             raise UnrecoverableError(response)
         return response.json()
 
+class ObserverCheck:
+    def __init__(self, host):
+        self._host = host
+
+    def latest_article(self, id):
+        template = "%s/report/latest-articles"
+        url = template % self._host
+        return _poll(
+            lambda: self._is_present(url, id),
+            "article with id %s at %s",
+            id, url
+        )
+
+    def _is_present(self, url, id):
+        response = requests.get(url)
+        LOGGER.debug("Loaded %s (%s)", url, response.status_code)
+        if response.status_code > 299:
+            raise UnrecoverableError(response)
+        soup = BeautifulSoup(response.content, "lxml-xml")
+        target_guid = "https://dx.doi.org/10.7554/eLife.%s" % id
+        guids = {item.guid.string:item for item in soup.rss.channel.find_all("item")}
```

Comment:
I'm a bit concerned by the length of the feed, which currently shows 8 articles in end2end. During the test suite 15-20 articles are published, sometimes in parallel, so with unlucky timing the one published in the new test could slip down and out of the list before it is seen here. Is there a pagination or a configuration we could set up observer with to make sure the list is long enough?

Reply:
the rss feeds are just reports using rss to serialise their results. all reports support pagination and changing the number of results returned. The reports are not part of the api raml but I'm using the same pagination parameters. Example: 
https://observer.elifesciences.org/report/latest-articles?per-page=1&page=20

most reports have multiple supported serialisations as well as a default serialisation. the default for latest and upcoming articles is rss, but they also support csv: 
https://observer.elifesciences.org/report/latest-articles?page=20&format=json